src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -684,8 +684,14 @@ public function dumpFile($filename, $content)
         // when the filesystem supports chmod.
         $tmpFile = $this->tempnam($dir, basename($filename));
 
-        if (false === @file_put_contents($tmpFile, $content)) {
-            throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null, $filename);
+        if (\is_resource($content)) {
+            $expectedSize = fstat($content)['size'] - ftell($content);
+        } else {
+            $expectedSize = array_sum(array_map('strlen', (array) $content));
```

Comment:
we don't want to add support for arrays - they worked "by chance" but that's not supported

Reply:
If we are not dealing with arrays in 3.4 it would break the current behavior – the one that is only explicitly deprecated in 4.0. That doesn’t seem the right thing to do.